packages/core/src/common/messaging/proxy-factory.ts

```diff
@@ -223,24 +225,27 @@ export class JsonRpcProxyFactory<T extends object> implements ProxyHandler<T> {
             return this.onDidCloseConnectionEmitter.event;
         }
         const isNotify = this.isNotification(p);
-        return (...args: any[]) =>
-            this.connectionPromise.then(connection =>
+        return (...args: any[]) => {
+            const method = p.toString();
+            const capturedError = new Error(`Request ${method} failed with message`);
```

Comment:
Wouldn't it be better to create the error only when needed, i.e. in the inner scope?

Reply:
The inner scope is always executed asynchronously and lose all information about the callee.

Reply:
Ah, the stacktrace is filled in the constructor. Makes sense.